File: testing/tests_js/test_031_form_element_loader.py
"""Node-backed checks for form-element module selection."""


# @features forms
# @dimensions invalid-schema
def test_unknown_form_element_reports_schema_type(run_node):
    run_node(
        r'''
import { getFormElement } from "./src/script/elements/loader.mjs";

try {
  await getFormElement(null, { type: "unsupported-field" }, null);
  throw new Error("Expected an unsupported form element to fail");
} catch (error) {
  if (error.message !== "Unknown form element type: unsupported-field") {
    throw error;
  }
}
''',
        module=True,
    )

File: testing/tests_js/test_031_form_element_loader.mjs
import assert from "node:assert/strict";
import { test } from "node:test";
import { getFormElement } from "../../src/script/elements/loader.mjs";

/** @pair forms:invalid-schema */
test("test_unknown_form_element_reports_schema_type", async () => {
	await assert.rejects(
		getFormElement(null, { type: "unsupported-field" }, null),
		{
			message: "Unknown form element type: unsupported-field",
		},
		"Unsupported form element did not report its schema type",
	);
});
